dsl/src/testground-runner.ts

```diff
@@ -0,0 +1,109 @@
+import { tmpdir } from 'os'
+import { promises as fs } from 'fs';
+import * as dsl from './dsl'
+import { transform } from './testground-transform'
+import path from 'path';
+import { execSync, spawn } from 'child_process';
+import { env } from 'process';
+
+function waitForTestground(testgroundHome: string) {
+    const execOpts = {
+        env: {
+            "TESTGROUND_HOME": testgroundHome,
+            "PATH": env["PATH"]
+        },
+        stdio: 'inherit'
+    } as const
+
+    console.log("Waiting for testground to be ready")
+    let tries = 10
+    // Wait for nc to make a connection
+    while (tries-- > 0) {
+        try {
+            execSync(`nc -z localhost 8042`, execOpts)
+        } catch {
+            execSync(`sleep 1`, execOpts)
+            continue
+        }
+        break
+    }
+
+    execSync(`testground healthcheck --runner local:docker --fix`, execOpts)
+}
+
+export async function run(testplans: dsl.TestPlans) {
+    const dir = path.join(tmpdir(), "testground")
+
+    // Check if directory exists
+    try {
+        await fs.access(dir)
+        await fs.rm(dir, { recursive: true, force: true })
+    } catch (e) {
+    }
+    await fs.mkdir(dir)
+    await fs.mkdir(path.join(dir, "testplans"))
+
+    const filesToCreate = transform(testplans)
+
+    await Promise.all(
+        filesToCreate.files.map(({ filename, filecontents }) =>
+            fs.writeFile(path.join(dir, "testplans", filename), filecontents)
+        )
+
+    )
+    await fs.writeFile(path.join(dir, ".env.toml"), `
+[daemon.scheduler]
+task_timeout_min = 2
```

Comment:
Would be better for this to be configured in the DSL rather than hardcoded here.